# Исследование надежности заемщиков


**Описание проекта**

Заказчик — кредитный отдел банка. Нужно разобраться, влияет ли семейное положение и количество детей клиента на факт погашения кредита в срок. Входные данные от банка — статистика о платёжеспособности клиентов.

Результаты исследования будут учтены при построении модели кредитного скоринга — специальной системы, которая оценивает способность потенциального заёмщика вернуть кредит банку.По представленным статистическим данным о платежеспособности клиентов банка необходимо провести исследование- влияет ли семейное положение и количество детей клиента на факт погашения кредита в срок. Результаты исследования будут учтены при построении модели кредитного скоринга — специальной системы, которая оценивает способность потенциального заёмщика вернуть кредит банку.

**Цель исследования — проверить гипотезы:**

Существует ли зависимость:

- между наличием детей и возвратом кредита в срок;
- между семейным положением и возвратом кредита в срок;
- между уровнем дохода и возвратом кредита в срок;
- между целью кредита и его возврат в срок.


**Предоставленные файлы**

`/datasets/data.csv` - данные о поведении пользователей (статистика о платёжеспособности клиентов)


**Основные шаги**

1. Обзор данных
2. Предобработка данных
3. Исследовательский анализ
4. Общий вывод

## Обзор данных

In [1]:
# импорт библиотек
import pandas as pd

Считаем CSV-файл с данными

In [2]:
try:
    data = pd.read_csv('https://code.s3.yandex.net//datasets/data.csv')
except:
    data = pd.read_csv('D:\myuser\YandexPracticumfilesproject\data (2).csv')

Выведем первые 20 строчек датафрейма `data` на экран.

In [3]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

В таблице 12 столбцов:
- **children** — количество детей в семье
- **days_employed** — общий трудовой стаж в днях
- **dob_years** — возраст клиента в годах
- **education** — уровень образования клиента
- **education_id** — идентификатор уровня образования
- **family_status** — семейное положение
- **family_status_id** — идентификатор семейного положения
- **gender** — пол клиента
- **income_type** — тип занятости
- **debt** — имел ли задолженность по возврату кредитов
- **total_income** — ежемесячный доход
- **purpose** — цель получения кредита

Выведем основную информацию о датафрейме с помощью метода `info()`.

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


В представленном для анализа файле - данные о поведении пользователей (статистика о платёжеспособности клиентов). Предварительно данных достаточно для анализа.

По полученной информации и первичном анализе видно следующее:
- В столбцах: (`days_employed,total_income`) имеются пропущенные значения. Придётся рассмотреть все столбцы и заполнить пропуски где это будет необходимо и возможно.
- Необходимо добавление столбцов `total_income_category` и `purpose_category` с категориями значений
- В столбце `total_income` неверный тип данных, в процессе предобработки изменим тип данных на формат целого числа int.
- Наименование столбцов стилистически верное.

## Предобработка данных

### Обработка пропусков

Выведем количество пропущенных значений для каждого столбца. Используем комбинацию двух методов.

In [5]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

In [6]:
data.loc[(data['days_employed'].isna()) & (data['total_income'].isna())]

children  days_employed  dob_years education  education_id  \
12            0            NaN         65   среднее             1   
26            0            NaN         41   среднее             1   
29            0            NaN         63   среднее             1   
41            0            NaN         50   среднее             1   
55            0            NaN         54   среднее             1   
...         ...            ...        ...       ...           ...   
21489         2            NaN         47   Среднее             1   
21495         1            NaN         50   среднее             1   
21497         0            NaN         48    ВЫСШЕЕ             0   
21502         1            NaN         42   среднее             1   
21510         2            NaN         28   среднее             1   

               family_status  family_status_id gender  income_type  debt  \
12          гражданский брак                 1      M    пенсионер     0   
26           женат / замужем                 0      M  госслужащий     0   
29     Не женат / не замужем                 4      F    пенсионер     0   
41           женат / замужем                 0      F  госслужащий     0   
55          гражданский брак                 1      F    пенсионер     1   
...                      ...               ...    ...          ...   ...   
21489        женат / замужем                 0      M    компаньон     0   
21495       гражданский брак                 1      F    сотрудник     0   
21497        женат / замужем                 0      F    компаньон     0   
21502        женат / замужем                 0      F    сотрудник     0   
21510        женат / замужем                 0      F    сотрудник     0   

       total_income                           purpose  
12              NaN                   сыграть свадьбу  
26              NaN                       образование  
29              NaN  строительство жилой недвижимости  
41              NaN  сделка с подержанным автомобилем  
55              NaN                   сыграть свадьбу  
...             ...                               ...  
21489           NaN              сделка с автомобилем  
21495           NaN                           свадьба  
21497           NaN        строительство недвижимости  
21502           NaN  строительство жилой недвижимости  
21510           NaN           приобретение автомобиля  

[2174 rows x 12 columns]

Мы обнаружили, что по столбцам days_employed,total_income одинаковое кол-во пропусков в 2174 ед. Вероятно имеется взаимосвязь наличия пропусков между данными колонками.

Столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.

In [7]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

Теперь заполним пропуски в столбце `days_employed` медианными значениями по каждому типу занятости income_type.

In [8]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

Убедимся, что все пропуски заполнены. Проверим себя и ещё раз выведем количество пропущенных значений для каждого столбца с помощью двух методов.

In [9]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Обработка аномальных значений

В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработаем значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.

In [10]:
data['days_employed'] = data['days_employed'].abs()

Для каждого типа занятости выведем медианное значение трудового стажа `days_employed` в днях.

In [11]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставим их как есть.

Выведем перечень уникальных значений столбца `children`

In [12]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

В столбце `children` есть два аномальных значения. Удалим строки, в которых встречаются такие аномальные значения из датафрейма `data`.

In [13]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

Ещё раз выведем перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.

In [14]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Изменение типов данных

Заменим вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.

In [15]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

Обработаем неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведём их к нижнему регистру.

In [16]:
data['education'] = data['education'].str.lower()

Выведем на экран количество явных дубликатов в данных. Если такие строки присутствуют, удалим их

In [17]:
data.duplicated().sum()

71

In [18]:
data = data.drop_duplicates()

### Категоризация данных

Для дальнейшего исследования, на основании диапазонов, указанных ниже по столбцу `total_income`, создадим  в датафрейме `data` столбец `total_income_category` с категориями:

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.

По данным категориям мы сможем укрупнить группы по ежемесячным доходам клиентов банка



Используем собственную функцию с именем `categorize_income()` и метод `apply()`.

In [19]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [32]:
data['total_income_category'] = data['total_income'].apply(categorize_income)
data['total_income_category'].value_counts()

total_income_category
C    15921
B     5014
D      349
A       25
E       22
Name: count, dtype: int64

Выведем на экран перечень уникальных целей взятия кредита из столбца `purpose`

In [21]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

Создадим функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.



In [22]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [23]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)
data.head(10)

children  days_employed  dob_years education  education_id  \
0         1    8437.673028         42    высшее             0   
1         1    4024.803754         36   среднее             1   
2         0    5623.422610         33   среднее             1   
3         3    4124.747207         32   среднее             1   
4         0  340266.072047         53   среднее             1   
5         0     926.185831         27    высшее             0   
6         0    2879.202052         43    высшее             0   
7         0     152.779569         50   среднее             1   
8         2    6929.865299         35    высшее             0   
9         0    2188.756445         41   среднее             1   

      family_status  family_status_id gender income_type  debt  total_income  \
0   женат / замужем                 0      F   сотрудник     0        253875   
1   женат / замужем                 0      F   сотрудник     0        112080   
2   женат / замужем                 0      M   сотрудник     0        145885   
3   женат / замужем                 0      M   сотрудник     0        267628   
4  гражданский брак                 1      F   пенсионер     0        158616   
5  гражданский брак                 1      M   компаньон     0        255763   
6   женат / замужем                 0      F   компаньон     0        240525   
7   женат / замужем                 0      M   сотрудник     0        135823   
8  гражданский брак                 1      F   сотрудник     0         95856   
9   женат / замужем                 0      M   сотрудник     0        144425   

                      purpose total_income_category          purpose_category  
0               покупка жилья                     B  операции с недвижимостью  
1     приобретение автомобиля                     C    операции с автомобилем  
2               покупка жилья                     C  операции с недвижимостью  
3  дополнительное образование                     B     получение образования  
4             сыграть свадьбу                     C        проведение свадьбы  
5               покупка жилья                     B  операции с недвижимостью  
6           операции с жильем                     B  операции с недвижимостью  
7                 образование                     C     получение образования  
8       на проведение свадьбы                     C        проведение свадьбы  
9     покупка жилья для семьи                     C  операции с недвижимостью

Столбцы с категориями успешно созданы

## Исследовательский анализ

#### Есть ли зависимость между количеством детей и возвратом кредита в срок?

Для начала обобщим  данные DataFrame `data` в виде сводной таблицы.

- 1.Найдём общую численность заёмщиков по категории кол-ва детей.
- 2.Найдём кол-во должников по категориям кол-ва детей.

In [24]:
data_pivot = data.pivot_table(index = ['children'], values = 'debt', aggfunc = ['count', 'sum','mean']) # Создадаим сводную таблицу методом .pivot_table()
data_pivot.columns = ['Кол-во заёмщиков','Кол-во должников','Доля должников'] # Переименуем многосоставные столбцы методом .columns['']
data_pivot.index.name = 'Кол-во детей' # Переименуем столбец(он находится в индексе) по которому группировали данные, методом .index.name =  ''
data_pivot # Вывод таблицы на экран


Кол-во заёмщиков  Кол-во должников  Доля должников
Кол-во детей                                                    
0                        14091              1063        0.075438
1                         4808               444        0.092346
2                         2052               194        0.094542
3                          330                27        0.081818
4                           41                 4        0.097561
5                            9                 0        0.000000

- 3.В сводной таблице осуществим арифметические преобразования по столбцу 'Доля должников' и отсортируем таблицу по столбцу Доля должников

In [25]:
data_pivot['Доля должников'] = (data_pivot['Доля должников'] *100).round(2).astype('str') # Переведём столбец в значения % в десятичную дробь с остатком до сотых 1. Для этого ужножем Series на 100 2.Методом .round() округлим значения Series до сотых 3.Изменим тип данных с float на str методом astype().
data_pivot['Доля должников']= data_pivot['Доля должников'] + '%' # Добавим %
data_pivot = data_pivot.sort_values(by = 'Доля должников') # Применим метод sort_values() для сортировки столбца 'Доля должников'
data_pivot # Выведем таблицу на экран

Кол-во заёмщиков  Кол-во должников Доля должников
Кол-во детей                                                   
5                            9                 0           0.0%
0                        14091              1063          7.54%
3                          330                27          8.18%
1                         4808               444          9.23%
2                         2052               194          9.45%
4                           41                 4          9.76%

 **Вывод**:  

 Благодаря нахождению процентного соотношения доли должников по категориям кол-ва детей, мы можем сделать следующие выводы:
- 1. Заемщик `не имеющий детей` самый ответственный и платёжеспособный, вероятность просрочки кредита 7.5 % по данной категории заёмщика.
- 2. Хуже обстоит дело с должниками, кто имеет `одного или двух детей`, здесь доля невозврата составляет соответственно 9,2% и 9,4%   


#### Есть ли зависимость между семейным положением и возвратом кредита в срок?

Для начала обобщим данные DataFrame data в виде сводной таблицы, используя методы: groupby() и agg()
- 1.Найдём общую численность заёмщиков по
категориям семейного положения.
- 2.Найдём кол-во должников по категориям семейного положения.

In [26]:
groupby_data = data.groupby('family_status').agg({'debt':['count','sum']})  # Создадаим сводную таблицу методом .groupby() и .agg()
groupby_data.columns = ['Кол-во заёмщиков','Кол-во должников']  # Переименуем многосоставные столбцы методом .columns['']
groupby_data.index.name = 'Семейное положение' # Переименуем столбец(он находится в индексе) по которому группировали данные методом .index.name =  ''
groupby_data # Вывод таблицы на экран

Кол-во заёмщиков  Кол-во должников
Семейное положение                                       
Не женат / не замужем              2796               273
в разводе                          1189                84
вдовец / вдова                      951                63
гражданский брак                   4134               385
женат / замужем                   12261               927

- 3.В сводной таблице создадим столбец 'Доля должников' для нахождения отношения численности должников к общей численности заёмщиков по категориям семейного положения.

In [27]:
groupby_data['Доля должников'] = groupby_data['Кол-во должников']/groupby_data['Кол-во заёмщиков'] #Создадим столбец и запишем в него отношения численности должников к общей численности заёмщиков
groupby_data['Доля должников'] = (groupby_data['Доля должников'] *100).round(2).astype('str') # Переведём столбец в значения % в десятичную дробь с остатком до сотых 1. Для этого ужножем Series на 100 2.Методом .round() округлим значения Series до сотых 3.Изменим тип данных с float на str методом astype().
groupby_data['Доля должников']= groupby_data['Доля должников'] + '%' # Добавим %
groupby_data = groupby_data.sort_values(by = 'Доля должников') # Применим метод sort_values() для сортировки столбца 'Доля должников'
groupby_data # выведем  таблицу на экран

Кол-во заёмщиков  Кол-во должников Доля должников
Семейное положение                                                      
вдовец / вдова                      951                63          6.62%
в разводе                          1189                84          7.06%
женат / замужем                   12261               927          7.56%
гражданский брак                   4134               385          9.31%
Не женат / не замужем              2796               273          9.76%

**Вывод:**

Благодаря нахождению процентного соотношения доли должников по категориям семейного положения, мы можем сделать следующие выводы:  
- 1. Заемщик с семейным положением `женат/замужем` достаточно ответственный и платёжеспособный, вероятность просрочки кредита 7.5 % по данной категории заёмщика.  
- 2. Хуже обстоит дело с должниками, кто находится в статусе семейного положения `гражданский брак`, вероятность просрочки кредита 9.3 % по данной категории заёмщика.
- 3. Заёмщик с семейным положением `Не женат/не замужем` составил антирейтинг возврата кредита в срок, вероятность просрочки кредита 9.7 % по данной категории заёмщика.


#### Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

Для начала обобщим  данные DataFrame 'data' в виде сводной таблицы, используя метод pivot_table.
- 1.Найдём общую численность заёмщиков по категории дохода.
- 2.Найдём кол-во должников по категории дохода.

In [28]:
data_pivot_income = data.pivot_table(index = ['total_income_category'], values = 'debt', aggfunc = ['count','sum']) # Создадаим сводную таблицу методом .pivot_table()
data_pivot_income.columns = ['Кол-во заёмщиков','Кол-во должников'] # Переименуем многосоставные столбцы методом .columns['']
data_pivot_income.index.name = 'Категория дохода заёмщиков' # Переименуем столбец(он находится в индексе) по которому группировали данные методом .index.name =  ''
data_pivot_income # Вывод таблицы на экран

Кол-во заёмщиков  Кол-во должников
Категория дохода заёмщиков                                    
A                                         25                 2
B                                       5014               354
C                                      15921              1353
D                                        349                21
E                                         22                 2

- 3.В сводной таблице создадим столбец 'Доля должников' для нахождения отношения численности должников к общей численности заёмщиков по категориям дохода.

In [29]:
data_pivot_income['Доля должников'] = data_pivot_income['Кол-во должников']/data_pivot_income['Кол-во заёмщиков'] #Создадим столбец и запишем в него отношения численности должников к общей численности заёмщиков
data_pivot_income['Доля должников'] = (data_pivot_income['Доля должников']*100).round(2).astype('str') # Переведём столбец в значения % в десятичную дробь с остатком до сотых 1. Для этого ужножем Series на 100 2.Методом .round() округлим значения Series до сотых 3.Изменим тип данных с float на str методом astype(). 4.Добавим %
data_pivot_income['Доля должников'] = data_pivot_income['Доля должников'] + '%'
data_pivot_income = data_pivot_income.sort_values(by = 'Доля должников') # Применим метод sort_values() для сортировки столбца 'Доля должников'
data_pivot_income # выведем  таблицу на экран

Кол-во заёмщиков  Кол-во должников Доля должников
Категория дохода заёмщиков                                                   
D                                        349                21          6.02%
B                                       5014               354          7.06%
A                                         25                 2           8.0%
C                                      15921              1353           8.5%
E                                         22                 2          9.09%

**Вывод:**

Благодаря нахождению процентного соотношения доли должников по категориям дохода заёмщиков, мы можем сделать следующие выводы:
- 1. Заемщики `категории B` (от 200001 до 1000000) одни из самых ответственных и платёжеспособных, вероятность просрочки кредита 7 % по данной категории заёмщика.
- 2. Заёмщики `категории С` (от 50001 до 200000) зачастую не выплачивают кредит в срок, вероятность просрочки кредита 8.5 % по данной категории заёмщика.


#### Как разные цели кредита влияют на его возврат в срок?

Для начала обобщим  данные DataFrame 'data' в виде сводной таблицы, используя методы: groupby() и agg().
- 1.Найдём общую численность заёмщиков по целям кредита.
- 2.Найдём кол-во должников по целям кредита.

In [30]:
groupby_data_purpose = data.groupby('purpose_category').agg({'debt':['count','sum']})  # Создадим сводную таблицу методом .groupby() и .agg()
groupby_data_purpose.columns = ['Кол-во заёмщиков','Кол-во должников']  # Переименуем многосоставные столбцы методом .columns['']
groupby_data_purpose.index.name = 'Цель кредита' # Переименуем столбец(он находится в индексе) по которому группировали данные методом .index.name =  ''
groupby_data_purpose # Вывод таблицы на экран

Кол-во заёмщиков  Кол-во должников
Цель кредита                                                
операции с автомобилем                4279               400
операции с недвижимостью             10751               780
получение образования                 3988               369
проведение свадьбы                    2313               183

- 3.В сводной таблице создадим столбец 'Доля должников' для нахождения отношения численности должников к общей численности заёмщиков по целям кредита.

In [31]:
groupby_data_purpose['Доля должников'] = groupby_data_purpose['Кол-во должников']/groupby_data_purpose['Кол-во заёмщиков'] #Создадим столбец и запишем в него отношения численности должников к общей численности заёмщиков
groupby_data_purpose['Доля должников'] = (groupby_data_purpose['Доля должников'] *100).round(2).astype('str') # Переведём столбец в значения % в десятичную дробь с остатком до сотых 1. Для этого ужножем Series на 100 2.Методом .round() округлим значения Series до сотых 3.Изменим тип данных с float на str методом astype().
groupby_data_purpose['Доля должников']= groupby_data_purpose['Доля должников'] + '%' # Добавим %
groupby_data_purpose = groupby_data_purpose.sort_values(by = 'Доля должников') # Применим метод sort_values() для сортировки столбца 'Доля должников'
groupby_data_purpose #groupby_data_purpose

Кол-во заёмщиков  Кол-во должников Доля должников
Цель кредита                                                               
операции с недвижимостью             10751               780          7.26%
проведение свадьбы                    2313               183          7.91%
получение образования                 3988               369          9.25%
операции с автомобилем                4279               400          9.35%

**Вывод:**

Благодаря нахождению процентного соотношения доли должников по целям кредита, мы можем сделать следующие выводы:
- 1. Заемщик с `ипотечным кредитом` самый  ответственный и исполнительный, вероятность просрочки кредита 7.2 % по данной категории заёмщика.Связано это с тем, что покупка недвижимости зачастую является жизненной необходимостью. Базовая потребность человека иметь собственный угол побуждает людей оформлять ипотеку, а страх конфискации побуждает исправно платить ежемесячные взносы.
- 2. Заёмщик с `автокредитом` составил антирейтинг возврата кредита в срок, вероятность просрочки кредита 9.3 % по данной категории заёмщика.
- 3. Примечательно, что болле половины всех кредитов были оформлены на цели связанные с недвижимостью.

## Общий вывод

**Отвечая на вопросы кредитного отдела банка, мы можем констатировать:**

**1.**
- Зависимость между количеством детей и возвратом кредита в срок определённо есть. Рассматривая самые многочисленные категории   заёмщиков было обнаружено, что каждый последующий ребёнок в семье увеличивает потенциальный риск просрочки платежа. В
  процентном соотношении это выглядит следующим образом:
-  `не имеющий детей 7.5 %`
-  `с одним ребёнком 9.2 %`
-  `с двумя детьми   9.4 %`


**2.**
- Зависимость между семейным положением и возвратом кредита в срок определённо есть. Рассматривая самые многочисленные категории заёмщиков было обнаружено, что заёмщики с  более 'серьёзным' статусом семейного положения, менее склонны просрочке платежа.В
  процентном соотношении это выглядит следующим образом:
-  `женат/замужем 7.5 %`
-  `гражданский брак 9.3 %`
-  `Не женат/не замужем 9.7%`

**3.**
- Зависимость между уровнем дохода и возвратом кредита в срок определённо есть. Рассматривая самые многочисленные категории заёмщиков было обнаружено, что заёмщики c более высоким доходом менее склонны просрочке платежа. В процентном соотношении это выглядит следующим образом:
-  `Заёмщики с уровнем дохода от 200001 до 1000000: 7 %`
-  `Заёмщики с уровнем дохода от 50001 до 200000: 8.5 %`

**4.**
- Зависимость между целями кредита и его возвратом в срок определённо есть. Рассматривая самые многочисленные категории заёмщиков было обнаружено, что заёмщики, котрые берут кредит на операции с недвижимостью  менее склонны к просрочке платежа. Страх остаться без жилья побуждает исправно платить ежемесячные взносы.  В процентном соотношении по остальным категориям  это выглядит следующим образом:
-  `операции с недвижимостью: 7.2 %`
-  `проведение свадьбы: 7.9 %`
-  `получение образования: 9.2%`
-  `операции с автомобилем: 9.3%`

**Рекомендации по улучшению системы кредитного скоринга**

1. В качестве рекомендаций  хотел бы посоветовать более внимательно относится к выгружаемой информацией в файлах. Произошла техническая ошибка при формировании таблицы с исходными данными, что привело к наличию пропусков в датасете. Стоит обозначить данную проблему отвественному лицу и попытаться предотвратить появление технических сбоев.
2. Для улучшения системы кредитного скоринга есть необходимость в добавлении дополнительных параметров и критерий оценки заемщика. Например: ежемесячный расход,должность,кол-во выданных ранее кредитов, сумма выданных кредитов.